src/mono/mono.proj

```diff
@@ -81,4 +86,64 @@
 
   <Target Name="Test" />
 
+  <!-- Copy Mono runtime bits to the local .dotnet dir for local experiments (temp solution) -->
+  <Target Name="PatchLocalDotnet" DependsOnTargets="ValidateTestHost">
+    <PropertyGroup>
+      <LocalDotnetRuntimeDir>$([System.IO.Directory]::GetDirectories("$(LocalDotnetDir)\shared\Microsoft.NETCore.App")[0])</LocalDotnetRuntimeDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <_MonoRuntimeArtifacts Include="$(BinDir)\*.*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(_MonoRuntimeArtifacts)"
+        DestinationFolder="$(LocalDotnetRuntimeDir)"
+        SkipUnchangedFiles="true" />
+  </Target>
+
+  <Target Name="PatchTestHost" DependsOnTargets="ValidateTestHost">
+    <ItemGroup>
+      <_MonoRuntimeArtifacts Include="$(BinDir)\*.*" />
+    </ItemGroup>
+    <Copy SourceFiles="@(_MonoRuntimeArtifacts)"
+          DestinationFolder="$(TestHostPath)\shared\Microsoft.NETCore.App\$(ProductVersion)" 
+          SkipUnchangedFiles="true"/>
+  </Target>
+
+  <!-- Run netcore\sample\HelloWorld sample using Mono Runtime -->
+  <Target Name="RunSample" DependsOnTargets="ValidateTestHost;PatchTestHost">
+    <MSBuild Projects="$(MonoProjectRoot)netcore\sample\HelloWorld\HelloWorld.csproj" Targets="Restore;Build" />
+    <Exec Command="$(TestHostPath)\$(DotNetExec) $(MonoProjectRoot)netcore\sample\HelloWorld\bin\HelloWorld.dll" />
+  </Target>
+
+  <!-- Run CoreCLR runtime test using testhost -->
+  <Target Name="RunCoreClrTest" DependsOnTargets="ValidateTestHost;PatchTestHost">
+    <Error Condition="$(CoreClrTest) == ''" Text="'CoreClrTest' is not set." />
+    <PropertyGroup>
+      <CORE_ROOT>$(TestHostPath)</CORE_ROOT>
+    </PropertyGroup>
+    <Exec Command="$(CoreClrTest)"/>
+  </Target>
+
+  <Target Name="RunCoreClrTests" DependsOnTargets="ValidateTestHost;PatchTestHost">
+    <PropertyGroup>
+      <CORE_ROOT>$(TestHostPath)</CORE_ROOT>
+      <ScriptExt Condition="'$(OS)' == 'Windows_NT'">.cmd</ScriptExt>
+      <ScriptExt Condition="'$(OS)' != 'Windows_NT'">.sh</ScriptExt>
+    </PropertyGroup>
+    <ItemGroup>
+      <CoreClrTests Include="$(ArtifactsDir)tests\coreclr\Windows_NT.x64.Checked\**\*$(ScriptExt)" />
```

Comment:
is the `Windows_NT.x64.Checked` used on non-Windows too?

Reply:
good point